privileges/privileges_linux.go

```diff
@@ -0,0 +1,34 @@
+//go:build linux || unix
+
+package privileges
+
+import (
+	"os"
+	"runtime"
+
+	"github.com/projectdiscovery/naabu/v2/pkg/israce"
+	"golang.org/x/sys/unix"
+)
+
+// isPrivileged checks if the current process has the CAP_NET_RAW capability or is root
```

Comment:
We need to also add a note in the comment that using this method will cause race detection to fail due to locking threads to CPUs